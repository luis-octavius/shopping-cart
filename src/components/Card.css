.card {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 300px;
    border-radius: 20px;
    padding: 10px;
    background-color: white;
    animation: width .6s ease;
    font-family: var(--font-secondary);
    transition: transform;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
}

.card:hover {
    transition: transform 125ms;
    transform: translateY(-10px);
    cursor: pointer;
}

.container-image {
    width: auto;
    height: 200px;
    position: relative;
}

.card-image {
    width: 100%;
    height: 100%;
}

.price-value {
    display: flex;
    justify-content: space-around;
    width: 100%;
    text-align: center;
}

.price-value h4 {
    font-size: 1.5rem;
}

.card h3 {
    font-size: 2rem;
    text-align: center;
    margin: 10px;
}

.add-button {
    width: 50px;
    bottom: 0;
    right: 0;
    animation: increase .4s ease-in-out;
}

.add-button:hover {
    transform: scale(1.4)
}

.highlight {
    display: block;
    text-align: center;
    color: var(--color-third);
}

.card-description {
    text-align: justify;
    font-size: 1rem;
    padding: 0 10px 0 10px;
}