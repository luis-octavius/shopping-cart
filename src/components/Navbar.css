.navbar {
    display: flex;
    justify-content: space-between;
    gap: 10px;
    width: 100%;
    height: 80px;
    font-family: var(--font-secondary);
    background-color: var(--color-primary);
    position: sticky;
}

.logo a {
    display: flex;
    height: 100%;
    text-decoration: none;
    cursor: pointer;
}

.logo a > h1 {
    margin: auto;
    color: var(--color-secondary);
    font-style: italic;
}

.shop-icon, .shop-link {
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
}

.shop-link > * {
    text-decoration: none;
    color: var(--color-secondary);
    font-size: 1.4rem;
    width: 200px;
    font-weight: 800;
}

.shop-icon {
    padding-right:20px;
}

.shop-icon img {
    height: 50px;
}

