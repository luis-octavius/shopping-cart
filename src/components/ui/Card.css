.card {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 400px;
    background-color: purple;
}

.card img {
    width: 350px;
}

.price-value {
    display: flex;
    justify-content: space-around;
    width: 95%;
}