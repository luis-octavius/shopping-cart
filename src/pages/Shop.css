.container {
    display: grid;
    width: 90%;
    margin: auto;
    height: auto;
    grid-auto-rows: auto;
    grid-template-columns: repeat(4, 1fr);
    gap: 20px;
    padding-top: 20px;
    justify-items: center;
}

.container-load {
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

.catalog {
    width: 25%;
    margin: auto;
    color: var(--color-primary);
    background-color: var(--color-fourth);
}

.catalog h1 {
    font-family: var(--font-secondary);
    text-align: center;
    font-size: 3rem;
    border: 4px solid var(--color-secondary);
}

@media (max-width: 600px) {
  .container { 
    grid-template-columns: repeat(2, 1fr); 
    }
}

@media (max-width: 900px) {
  .container { 
    grid-template-columns: repeat(3, 1fr); 
    }
}
