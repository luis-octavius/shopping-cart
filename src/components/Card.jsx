import "./Card.css";
import shopAdd from "../assets/shopping-cart-add.svg";
import useStore from "../utils/store";

export default function Card({ product }) {
  const { increment, addProduct } = useStore();

  return (
    <div className="card">
      <h3>{product.title}</h3>
      <div className="container-image">
        <img className="card-image" src={product.image} />
      </div>
      <div className="price-value">
        <h4>
          <span className="highlight">Price</span>R$ {product.price}
        </h4>
        <h4>
          <span className="highlight">Rating</span>
          {product.rating.rate}
        </h4>
        <img
          className="add-button"
          src={shopAdd}
          alt="cart with plus sign"
          aria-label="Add to Cart"
          data-testid="add-button"
          onClick={() => {
            increment();
            addProduct(product.id);
          }}
        />
      </div>
      <div className="card-description">
        <p>{product.description}</p>
      </div>
    </div>
  );
}
