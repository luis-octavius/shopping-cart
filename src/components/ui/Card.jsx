/*
Descrição
Imagem
Preço
Avaliação
Título
*/
import './Card.css';

export default function Card({product}) {
    return (
        <div className="card">
            <h3>{product.title}</h3> 
            <img src={product.img} />
            <div className="price-value">
                <h4>{product.price}</h4>
                <h4>{product.rating}</h4>
            </div>
            <div>
                <p>{product.description}</p>
            </div>
        </div>
    )
}