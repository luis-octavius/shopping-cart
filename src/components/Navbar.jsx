import './Navbar.css'
import shopImg from '../assets/shopping.svg';
import shopBagImg from '../assets/shopping-bag.svg';
import { Link } from 'react-router';
import { useEffect, useState } from 'react';

export default function Navbar() {
    const [quantity, setQuantity] = useState(0);

    return (
        <div className="navbar shadow">
            <div className="logo">
                <Link to="/home" className="link">
                    <img src={shopBagImg} />
                    <h1>Shoppex</h1>
                </Link>
            </div>

            <div className="shop-link">
                <Link to="/shop">
                Shop
                </Link>
            </div>

            <div className="shop-icon">
                <Link to="/cart" className="cart-shop">
                    <img src={shopImg} />
                    {quantity !== true && (
                        <div className="items">{quantity}</div>
                    )}
                </Link>
            </div>
        </div>
    )
}