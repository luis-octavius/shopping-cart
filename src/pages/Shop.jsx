import { 
    useState, 
    useEffect,
    useContext
} from 'react';
import { UserContext } from '../utils/UserContext';
import PropTypes from 'prop-types';
import Card from '../components/Card';
import './Shop.css'
import Loading from '../components/ui/Loading';
import useStore from '../utils/store';

export default function Shop() {
    const data = useContext(UserContext);

    const [isLoading, setIsLoading] = useState(true);
    const { count } = useStore();

    useEffect(() => {
            setIsLoading(false);
    },[isLoading, count]);

    if (isLoading) return (
        <div className="container-load">
            <Loading />
        </div>
    )

    return (
        <>
            <div className="catalog">
                <h1>Catalog</h1>
            </div>
        <div className="container">
            {data && data.map((item, index) => (
                <Card 
                    product={item}
                    key={index}    
                />
            ))}
        </div>
        </>
    )
}

Shop.propTypes = {
    count: PropTypes.number.isRequired,
}