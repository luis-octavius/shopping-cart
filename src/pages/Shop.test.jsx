import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '../utils/test-utils';
import userEvent from '@testing-library/user-event';
import Shop from './Shop.jsx';
import Navbar from '../components/Navbar';
import Card from '../components/Card';

describe("Shop", () => {
    const mockProducts = {
        id: 1,
        description: "A simple product blablabla",
        image: "https://picsum.photos/200/300",
        title: "Product",
        rating: {
            rate: 4.5, 
        },
        price: 99.99
    }

    it("renders Shop", () => {
        const { container } = render(
        <Shop/>, {
            userValue: mockProducts,
            storeState: { count: 0 }
        });

        expect(container).toMatchSnapshot();
    })

    
})