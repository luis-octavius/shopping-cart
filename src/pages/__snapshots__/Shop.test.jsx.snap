// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > Renders Shop 1`] = `
<div>
  <div
    class="catalog"
  >
    <h1>
      Catalog
    </h1>
  </div>
  <div
    class="container"
  />
</div>
`;
